src/foundation/variables.css

```diff
@@ -1 +1,180 @@
-/* stylelint-disable no-empty-source */
+/* Variables
+ *
+ * Variables should follow the `--component-state-property-size` formula for
+ * consistent naming.
+ * Ex: var(--nav-link-disabled-color) and --modal-content-box-shadow-xs.
+
+/* stylelint-disable custom-property-empty-line-before */
+
+/* Color system */
+
+:root {
+  --white: #fff;
+  --black: #000;
+  --gray-50: #fafafa;
+  --gray-100: #f5f5f5;
+  --gray-200: #eee;
+  --gray-300: #e0e0e0;
+  --gray-400: #bdbdbd;
+  --gray-500: #9e9e9e;
+  --gray-600: #757575;
+  --gray-700: #616161;
+  --gray-800: #424242;
+  --gray-900: #212121;
+
+  /* Spacing
+   *
+   * Control the default styling of most Bootstrap elements by modifying these
+   * variables. Mostly focused on spacing.
+   */
+
+  --spacer: 1.5rem;
+
+  /* Body
+   *
+   * Settings for the `<body>` element.
+   */
+  --body-bg: var(--white);
+  --body-color: var(--gray-900);
+  --body-text-align: initial;
+
+  /* Links
+   *
+   * Style anchor elements.
+   */
+  --link-color: var(--primary);
+  --link-decoration: none;
+  --link-hover-color: var(--primary-dark);
+  --link-hover-decoration: underline;
+
+  /* Darken percentage for links with `.text-*` class (e.g. `.text-success`) */
+  --emphasized-link-hover-darken-percentage: 15%;
+  --stretched-link-pseudo-element: after;
+  --stretched-link-z-index: 1;
+
+  /* Paragraphs
+   *
+   * Style p element.
+   */
+
+  --paragraph-margin-bottom: var(--spacer);
+
+  /* Components
+   *
+   * Define common padding and border radius sizes and more.
+   */
+
+  --border-width: 1px;
+  --border-color: var(--gray-300);
+  --border-radius: 0.25rem;
+  --border-radius-lg: 0.3rem;
+  --border-radius-sm: 0.2rem;
+  --rounded-pill: 50rem;
+  --box-shadow-sm: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
+  --box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
+  --box-shadow-lg: 0 1rem 3rem rgba(0, 0, 0, 0.175);
+  --box-shadow-inset: inset 0 1px 2px rgba(0, 0, 0, 0.075);
+  --component-active-color: var(--white);
+  --component-active-bg: var(--primary);
+  --caret-width: 0.3em;
+  --caret-vertical-align: calc(var(--caret-width) * 0.85);
+  --caret-spacing: calc(var(--caret-width) * 0.85);
```

Comment:
@Joozty I am in order to remove this but I am not really sure about.

What do you think about [Bootstrap carets](https://github.com/twbs/bootstrap/blob/b2ae6a6a94a70a027469fa64723e8268ca66254b/scss/mixins/_caret.scss)? Is it good idea?

They use old-fashion border trick when you have 2 transparent borders and 1 coloured, it draws a triangle. It's a zero-dependency carets what I like but on the other hand it's little bit tricky what is confusing me.

I think I will remove and add it back when needed. We will need to support mixing for this feature.

Reply:
Yesterday I discovered how those carets work. It took me some time to understand it. 😄 

I this trick [in dashboard search bar](https://github.com/oacore/dashboard/blob/layout/components/searchbar/index.scss#L10-L20). I guess it's of having it in the design library. However, we can postpone it. :)